src/NUnitFramework/framework/Exceptions/MultipleAssertException.cs

```diff
@@ -36,9 +36,21 @@ namespace NUnit.Framework
     public class MultipleAssertException : ResultStateException
     {
         /// <summary>
-        /// Default Constructor (normally used)
+        /// Default Constructor
         /// </summary>
-        public MultipleAssertException() : base("One or more failures in Multiple Assert block:") { } 
+        public MultipleAssertException(string message) : base(message) { }
+
+        /// <summary>
+        /// Construct based on the TestResult so far. This is the constructor
+        /// used normally, when exiting the multiple assert block with failures.
+        /// Not used internally but provided to facilitate debugging.
+        /// </summary>
+        /// <param name="result">The current result</param>
+        public MultipleAssertException(ITestResult result)
+            : base(result.Message)
```

Comment:
Missing argument validation. If `result` is null, this will throw. You'll probably need some inline static validation to keep using it as an argument to the base. Something like this:
```
: base(EnsureNotNull(result).Message)
```

Reply:
> Maybe base(result?.Message) and a Guard in the constructor.

That works but the validation will be misleading as you'd be validating the internal `Message` argument and not the `result` itself. A static validation check in this class can properly indicate that `result` itself is `null`.